python/lsst/ip/isr/isrMockLSST.py

```diff
@@ -0,0 +1,597 @@
+        self.config.doAddFlat = False
+
+
+# Classes to generate calibration products mocks.
+class DarkMockLSST(ReferenceMockLSST):
+    """Simulated reference dark calibration.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doAddDark = True
+        self.config.darkTime = 1.0
+
+
+class BiasMockLSST(ReferenceMockLSST):
+    """Simulated combined bias calibration.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doAddBias = True
+        self.config.doAddGain = True
+        self.config.readNoise = 10.0
+
+
+class FlatMockLSST(ReferenceMockLSST):
+    """Simulated reference flat calibration.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doAddFlat = True
+
+
+class FringeMockLSST(ReferenceMockLSST):
+    """Simulated reference fringe calibration.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doAddFringe = True
+
+
+class BfKernelMockLSST(IsrMockLSST):
+    """Simulated brighter-fatter kernel.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doGenerateImage = False
+        self.config.doGenerateData = True
+
+        # calibration products configs
+        self.config.doBrighterFatter = True
+        self.config.doDefects = False
+        self.config.doCrosstalkCoeffs = False
+        self.config.doTransmissionCurve = False
+
+
+class DefectMockLSST(IsrMockLSST):
+    """Simulated defect list.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doGenerateImage = False
+        self.config.doGenerateData = True
+
+        self.config.doBrighterFatter = False
+        self.config.doDefects = True
+        self.config.doCrosstalkCoeffs = False
+        self.config.doTransmissionCurve = False
+
+
+class CrosstalkCoeffMockLSST(IsrMockLSST):
+    """Simulated crosstalk coefficient matrix.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doGenerateImage = False
+        self.config.doGenerateData = True
+
+        self.config.doBrighterFatter = False
+        self.config.doDefects = False
+        self.config.doCrosstalkCoeffs = True
+        self.config.doTransmissionCurve = False
+
+
+class TransmissionMockLSST(IsrMockLSST):
+    """Simulated transmission curve.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doGenerateImage = False
+        self.config.doGenerateData = True
+
+        self.config.doBrighterFatter = False
+        self.config.doDefects = False
+        self.config.doCrosstalkCoeffs = False
+        self.config.doTransmissionCurve = True
+
+
+class MockLSSTDataContainer(object):
+    """Container for holding ISR LSST mock objects.
+    """
+    dataId = "isrMockLSST Fake Data"
+    darkval = 2.  # e-/sec
```

Comment:
These should derive from the config options, but I also don't recall how they're used, so it's probably fine for now.

Reply:
I am removing the containers as they are out of date and I don't see their use at the moment.